docs/painless/painless-basic-syntax.asciidoc

```diff
@@ -0,0 +1,304 @@
+[[painless-basic-syntax]]
+=== Basic Syntax
+
+[[comments]]
+==== Comments
+
+Painless supports both single-line and multi-line comments. Comments can be
+included anywhere within a script. Use the `//` token anywhere on a line to
+specify a single-line comment. All characters from the `//` token to the end
+of the line are ignored. Use an opening `/*` token and a closing `*/` token
+to specify a multi-line comment. Multi-line comments can start anywhere on a
+line, and all characters in between the `/*` token and `*/` token are ignored.
+
+*Grammar:*
+[source,ANTLR4]
+----
+SINGLE_LINE_COMMENT: '//' .*? [\n\r];
+MULTI_LINE_COMMENT: '/*' .*? '*/';
+----
+
+*Examples:*
+
+Single-line and multi-line comments where `<code>` represents tokens within a
+script.
+
+[source,Java]
+----
+// single-line comment
+
+<code> // single-line comment
+
+/* multi-
+   line
+   comment */
+
+<code> /* multi-line
+          comment */ <code>
+
+<code> /* multi-line comment */ <code>
+----
+
+[[keywords]]
+==== Keywords
+
+The keywords in the table below are reserved for built-in language
+features. These keywords cannot be used as identifiers or types.
+
+[cols="^1,^1,^1,^1,^1"]
+|====
+| if | else | while | do | for
+| in | continue | break | return | new
+| try | catch | throw | this | instanceof
+|====
+
+[[literals]]
+==== Literals
+
+Use literals to specify different types of values directly in a script.
+
+[[integers]]
+===== Integers
+
+Use integer literals to specify an integer value in decimal, octal, or hex
+notation of the <<primitive-types, primitive types>> int, long, float, or
+double. Use the following single letter designations to specify the
+<<primitive-types, primitive type>>: `l` or `L` for `long`, `f` or `F` for
+`float`, and `d` or `D` for `double`. If not specified, the type defaults to
+`int`.  Use `0` as a prefix to specify an integer literal as octal, and use
+`0x` or `0X` as a prefix to specify an integer literal as hex.
+
+*Grammar:*
+[source,ANTLR4]
+----
+INTEGER: '-'? ( '0' | [1-9] [0-9]* ) [lLfFdD]?;
+OCTAL:   '-'? '0' [0-7]+ [lL]?;
+HEX:     '-'? '0' [xX] [0-9a-fA-F]+ [lL]?;
+----
+
+*Examples:*
+
+Integer literals of `int 0`, `double 0.0`, `long 1234`,
+`float -90.0`, `int -18` in octal, and `int 3882` in hex.
+
+[source,Java]
```

Comment:
I'd probably use `[source,painless]` here instead. The language doesn't really do anything at documentation rendering time. It is used more when we test the documentation. And it might be useful to have this be accurate.

Reply:
Done.  Will work through switching all the examples as I edit those sections.